core/client/fs/src/main/java/alluxio/client/metrics/MetricsHeartbeatContext.java

```diff
@@ -52,6 +59,7 @@
 @ThreadSafe
 public class MetricsHeartbeatContext {
   private static final Logger LOG = LoggerFactory.getLogger(MetricsHeartbeatContext.class);
+  private static final long SHUTDOWN_TIMEOUT_MS = 10 * Constants.SECOND_MS;
```

Comment:
10 seconds seems like a long shutdown timeout to me. For metrics, I would prefer it to have as little impact as possible on the shutdown of the JVM. Maybe 2-3 seconds is better?